rts/motoko-rts/src/idl.rs

```diff
@@ -52,7 +53,11 @@ pub unsafe fn leb128_decode_ptr(buf: *mut Buf) -> (u32, *mut u8) {
 }
 
 unsafe fn is_primitive_type(ty: i32) -> bool {
-    ty < 0 && (ty >= IDL_PRIM_lowest || ty == IDL_REF_principal || ty == IDL_EXT_region)
+    ty < 0
+        && (ty >= IDL_PRIM_lowest
+            || ty == IDL_REF_principal
+            || ty == IDL_EXT_region
+            || ty == IDL_EXT_blob)
```

Comment:
Added Blob for stable types only, cool! Though I wonder if we now accept (pure Candid) type tables that contain IDL_EXT_Blob maliciously.
We need to check if is_primitive is used during type table validation.

Ok. I checked the original and modifying this definition seems to break candid validation (`parse_idl_header`)
Maybe just define a second, Blob including version for use in memory_check?

Reply:
Good catch, thanks a lot. This is very relevant. I assumed in error that we only serialize Candid on our own in the compiler but this is not true as we receive external candid-binary too that we parse. I will fix this.